enhancements/audit-command.md

```diff
@@ -0,0 +1,722 @@
+ 
+**IMPORTANT**: Note that we are describing here an analytic tool/command that will start to be implemented in **alpha**. This feature uses what is in place and implemented already and cover needs which are raised in other proposals such as [Custom operator validation #43](https://github.com/operator-framework/enhancements/pull/43). However, it does not make any change or requires any new behaviour for the operator authors. The possibilities covered by the audit command can be easily documented after the first version of it this command be in place. 
+
+## Proposal
+
+Add a new command `audit` with the purpose to audit all desired operator bundle data of a specific index catalog. 
+
+By default, the command output a report where it is possible to have a view of the main characteristics and aspect of all operator bundles shipped in a specified OLM index catalog as check if they are passing in criteria and checks required to ensure their quality. However, accordingly with the purpose of its usage is also possible customize what should or not be done by this operation. 
+
+This command will perform checks by using the [SDK](https://github.com/operator-framework/operator-sdk) available sub commands; `bundle validate` and `scorecard bundle`.
+
+In this way, the binary will provide the alpha audit sub-command which would like: 
+
+```
+$ alpha audit --help
+
+Flags:
+  -h, --help                     help for audit
+  -i, --index-image              index image to get package from. 
+  -c, --container-tools          the container tool used to login in the registry and pull and extract the images. One of: [docker, podman] (default "docker")
+  -w, --wait-time duration       minutes to wait for the report to be complete. Example: 60m (default timeout is 30m)       
+  -o, --output                   output format for results. Valid values: xlsx, json (default "xlsx")
+  -s, --scorecard bool           set false to disable the scorecard checks. (default true). Note that scorecard test requires a cluster running. 
+  -v, --validation bool          set false to disable the bundle validation checks. (default true). 
+  -d, --data                     set false to disable the data report. It is useful for who would like to audit the index via only run the bundle and scorecard checks. (default true).
+  -h, --head-only                set the flag to audit only the operator bundles which are the head of the channel. (default all bundles of the index catalog image will be used)
+  -w, --warning                  set false to disable the report of warning(s) found in the checks (default warning and errors will be reported)
+```
+
+Note that the command will:
+- Use the index image to gathering the data
+- Download the operator bundles
+- Use SDK commands to run the checks 
+- Output the results
+
+**NOTE** If we move forward with this command as part of opm it would be like: `opm alpha index audit --help`
```

Comment:
I think `alpha index audit` is a WAY better command.

Reply:
done

Reply:
If it is addressed in opm yes it might make sense. However, if it is a new tool for example then it shows not appropriated any more. IMO which is relevant here is we defined how and what does audit command then, this kind of detail shows not matter too much as to where it will be implemented.